libkbfs/md_journal.go

```diff
@@ -652,12 +679,36 @@ func (j *mdJournal) put(
 
 	mStatus := rmd.MergedStatus()
 
-	if (mStatus == Unmerged) && (rmd.BID() == NullBranchID) {
-		j.log.CDebugf(
-			ctx, "Changing branch ID to %s and prev root to %s for MD for TLF=%s with rev=%s",
-			lastBranchID, lastMdID, rmd.TlfID(), rmd.Revision(), rmd.BID())
-		rmd.SetBranchID(lastBranchID)
-		rmd.SetPrevRoot(lastMdID)
+	if mStatus == Unmerged {
```

Comment:
It's ok with me.  You could break up each of the sections into a helper function if you wanted it to look a bit cleaner (this block here could be something like `prepareUnmergedPut(rmd, lastBranchID, lastMdID head) (allowedReplace, error)`).  But that wouldn't really simplify the logic per se.  I don't have any great ideas for that.